cmake/Modules/CocosBuildHelpers.cmake

```diff
@@ -169,27 +169,57 @@ function(cocos_mark_code_files cocos_target)
   endif()
 
   message(STATUS "cocos_mark_code_files: ${cocos_target}")
-  set(group_base "Source Files")
 
   get_property(file_list TARGET ${cocos_target} PROPERTY SOURCES)
 
+  get_headers_by_sources(app_headers CC_SOURCES ${file_list})
+  list(APPEND file_list ${app_headers})
   foreach(single_file ${file_list})
-    # get relative_path
-    get_filename_component(abs_path ${single_file} ABSOLUTE)
-    file(RELATIVE_PATH relative_path_with_name ${root_dir} ${abs_path})
-    get_filename_component(relative_path ${relative_path_with_name} PATH)
-    # set source_group, consider sub source group 
-    string(REPLACE "/" "\\" ide_file_group "${group_base}/${relative_path}")
-    source_group("${ide_file_group}" FILES ${single_file})
+    source_group_single_file(${single_file} GROUP_TO "Source Files" BASE_PATH "${root_dir}")
+  endforeach()
+
+endfunction()
+
+# get headers from `CC_SOURCES` save result in `headers_out`
+function(get_headers_by_sources headers_out)
```

Comment:
I think that we have already too many function... It is too complicated. 
Consider use GLOB_RECURSE for headers
https://cmake.org/cmake/help/latest/command/file.html
It is not recommended for src files but for header files I think it is fine.

Reply:
I considered it before. but when we `GLOB_RECURSE` headers, we will get all platform headers, useless headers will be collected too for a specific build.

yes, too many `function`  is complicated, but `function` is more clear than `macro` for cmake, I will have to write much duplicate code if I don't use `function` or `macro`.

Reply:
Than maybe it will be simpler to just have list of headers?
Any way this function may not find averything...

Reply:
You are right, we should write the header list for headers. it's the best choice.

write header list will spend me more than 2 days and it's a little boring... Ok, I will do it in next week when I'm free.

Reply:
It should not be so boring with i.e. "find -name *.h"